terrain/voxel_block.cpp

```diff
@@ -7,27 +7,32 @@
 // Faster version of Mesh::create_trimesh_shape()
 // See https://github.com/Zylann/godot_voxel/issues/54
 //
-static Ref<ConcavePolygonShape> create_concave_polygon_shape(Array surface_arrays) {
+static Ref<ConcavePolygonShape> create_concave_polygon_shape(Vector<Array*> surfaces) {
 
-	PoolVector<Vector3> positions = surface_arrays[Mesh::ARRAY_VERTEX];
-	PoolVector<int> indices = surface_arrays[Mesh::ARRAY_INDEX];
+	PoolVector<Vector3> face_points;
 
-	ERR_FAIL_COND_V(positions.size() < 3, Ref<ConcavePolygonShape>());
-	ERR_FAIL_COND_V(indices.size() < 3, Ref<ConcavePolygonShape>());
-	ERR_FAIL_COND_V(indices.size() % 3 != 0, Ref<ConcavePolygonShape>());
+	for(int i = 0; i < surfaces.size(); i++) {
+		const Array &surface_arrays = *surfaces[i];
 
-	int face_points_count = indices.size();
+		PoolVector<Vector3> positions = surface_arrays[Mesh::ARRAY_VERTEX];
+		PoolVector<int> indices = surface_arrays[Mesh::ARRAY_INDEX];
 
-	PoolVector<Vector3> face_points;
-	face_points.resize(face_points_count);
+		ERR_FAIL_COND_V(positions.size() < 3, Ref<ConcavePolygonShape>());
+		ERR_FAIL_COND_V(indices.size() < 3, Ref<ConcavePolygonShape>());
+		ERR_FAIL_COND_V(indices.size() % 3 != 0, Ref<ConcavePolygonShape>());
 
-	{
-		PoolVector<Vector3>::Write w = face_points.write();
-		PoolVector<int>::Read index_r = indices.read();
-		PoolVector<Vector3>::Read position_r = positions.read();
+		int face_points_count = indices.size();
+		int face_points_start = face_points.size();
+		face_points.resize(face_points_start + face_points_count);
 
-		for (int i = 0; i < face_points_count; ++i) {
-			w[i] = position_r[index_r[i]];
+		{
+			PoolVector<Vector3>::Write w = face_points.write();
+			PoolVector<int>::Read index_r = indices.read();
+			PoolVector<Vector3>::Read position_r = positions.read();
+
+			for (int i = face_points_start; i < face_points_count; ++i) {
+				w[i] = position_r[index_r[i]];
```

Comment:
When this code gets to the second surface, `i` will no longer start from the beginning of the surface, resulting in buffer overrun. You should probably use a second counter variable.